Allow configuring Elasticsearch cluster name

dnhatn opened the pull request:
To support multiple clusters, we need the ability to specify the cluster name when installing an Elasticsearch instance. This is not strictly required, but it prevents Elasticsearch nodes from joining the wrong cluster.

pquentin reviewed (changes requested):
Can you please look into the lint failures? They're real issues, eg. `DockerProvisioner` is not instantiated with `cluster_name` in `esrally/mechanic/provisioner.py`

Have you tried running the tests locally?

dnhatn commented:
@pquentin I ran `make test` but I didn't see any failure. I will look into the lint failure. Thanks for looking.

dnhatn commented:
@pquentin Great, thank you for the pointer 👍.

dnhatn commented:
@pquentin It's ready for another round. Can you take another look? Thank you!

pquentin commented on esrally/rally.py:
```diff
@@ -400,6 +400,11 @@ def add_track_source(subparser):
+        "--cluster-name",
+        help="The name of this Elasticsearch cluster",
+        default="rally-benchmark",
+    )
```
Can you please add this option to the `race` command too and document it in https://esrally.readthedocs.io/en/stable/command_line_reference.html#command-line-flags?

pquentin reviewed (changes requested):
Thanks!

Can you please add a test? I think we can add `--cluster-name={cfg}` to `TestCluster.install` and then at the end of `TestCluster.start` use the `es` client to get the cluster name and `assert` that it is `cfg`. This won't test all combinations but should be good enough.

dnhatn commented on esrally/mechanic/provisioner.py:
```diff
@@ -31,26 +31,30 @@
 
 def local(cfg, car, plugins, ip, http_port, all_node_ips, all_node_names, target_root, node_name):
     distribution_version = cfg.opts("mechanic", "distribution.version", mandatory=False)
+    cluster_name = cfg.opts("mechanic", "cluster.name", default_value="rally=benchmark", mandatory=False)
```
So sorry for this @pquentin.

dnhatn commented:
@pquentin Thanks for reviews. I've addressed your comments. Would you please take another look?

pquentin reviewed (changes requested):
Thanks! There's mostly only documentation left.

dnhatn commented on docs/cluster_management.rst:
```diff
@@ -67,11 +67,11 @@ Levelling Up: Benchmarking a Cluster
+    export INSTALLATION_ID=$(esrally install --quiet --distribution-version=8.1.0 --node-name="rally-node-1" --cluster-name=rally-benchmark --network-host="192.168.14.78" --http-port=39200 --master-nodes="rally-node-0,rally-node-1,rally-node-2" --seed-hosts="192.168.14.77:39300,192.168.14.78:39300,192.168.14.79:39300" | jq --raw-output '.["installation-id"]')
     # on 192.168.14.79
-    export INSTALLATION_ID=$(esrally install --quiet --distribution-version=7.4.2 --node-name="rally-node-2" --network-host="192.168.14.79" --http-port=39200 --master-nodes="rally-node-0,rally-node-1,rally-node-2" --seed-hosts="192.168.14.77:39300,192.168.14.78:39300,192.168.14.79:39300" | jq --raw-output '.["installation-id"]')
+    export INSTALLATION_ID=$(esrally install --quiet --distribution-version=8.1.0 --node-name="rally-node-2" --cluster-name=rally-benchmark --network-host="192.168.14.79" --http-port=39200 --master-nodes="rally-node-0,rally-node-1,rally-node-2" --seed-hosts="192.168.14.77:39300,192.168.14.78:39300,192.168.14.79:39300" | jq --raw-output '.["installation-id"]')
```
Ok, I reverted changes in this file.

dnhatn commented on docs/command_line_reference.rst:
```diff
@@ -272,6 +272,15 @@ A unique identifier for an installation. This id is automatically generated by R
+
+    This command line parameter is experimental. Expect the functionality and the command line interface to change significantly even in patch releases.
+
+Used to specify the cluster name of the current node when it is setup via the ``install`` subcommand.
```
+1, I expanded the document for this parameter in https://github.com/elastic/rally/pull/1478/commits/6946dd077fd7dc8d1b94639438cd73d2c5a69d30

dnhatn commented on docs/command_line_reference.rst:
```diff
@@ -272,6 +272,15 @@ A unique identifier for an installation. This id is automatically generated by R
+
+.. warning::
+
+    This command line parameter is experimental. Expect the functionality and the command line interface to change significantly even in patch releases.
```
Yes, because of the surrounding parameters. I can remove it if you prefer.

dnhatn commented:
@pquentin Sorry for the delay. I switched my focus to other topics recently. I have addressed your comments. Would you mind taking another look? Thank you!

inqueue commented on docs/command_line_reference.rst:
```diff
@@ -272,6 +272,26 @@ A unique identifier for an installation. This id is automatically generated by R
+
+    This command line parameter is experimental. Expect the functionality and the command line interface to change significantly even in patch releases.
+
+This parameter is useful in benchmarks involved multiple Elasticsearch clusters. It's used to configure the cluster name of the current Elasticsearch node when it is setup via the ``install`` or ``race`` subcommand. The following example setups two Elasticsearch clusters: ``cluster-1`` and ``cluster-2``, and each has two nodes::
```
Nit: change "setups" to "sets up".

dnhatn replied:
thanks, fixed in https://github.com/elastic/rally/pull/1478/commits/538418d0e19670e7efcf6d6e8528c8cc3eb4ffe3

pquentin reviewed (approved):
Thanks for iterating! LGTM.
